src/ray/gcs/store_client/in_memory_store_client.cc

```diff
@@ -127,21 +137,28 @@ Status InMemoryStoreClient::AsyncBatchDelete(const std::string &table_name,
 }
 
 int InMemoryStoreClient::GetNextJobID() {
-  absl::MutexLock lock(&mutex_);
-  job_id_ += 1;
-  return job_id_;
+  return job_id_.fetch_add(1, std::memory_order_acq_rel);
```

Comment:
What if we use sequential order? Curious on the performance impact?

Reply:
sequential enforces strict single order, acq rel allows some reordering so possible to be slightly more performant, but could actually be the same in this case

Reply:
Yeah I know, I'm just curious on the performance improvement.